HELP.md

```diff
@@ -0,0 +1,761 @@
+
+E.g.
+
+```rust
+#[near_bindgen]
+impl Contract {
+    pub fn withdraw_100(&mut self, receiver_id: ValidAccountId) -> Promise {
+        Promise::new(receiver_id.into()).transfer(100)
+    }
+}
+```
+
+## Use high-level cross-contract API
+
+There is a helper macro that allows to make cross-contract calls called `#[ext_contract(...)]`. It takes a Rust Trait and
+converts it to a module with static methods. Each of these static methods takes positional arguments defined by the Trait,
+then the `receiver_id`, the attached deposit and the amount of gas and returns a new `Promise`.
+
+For example, let's define a calculator contract Trait:
+
+```rust
+#[ext_contract(ext_calculator)]
+trait Calculator {
+    fn mult(&self, a: U64, b: U64) -> U128;
+
+    fn sum(&self, a: U128, b: U128) -> U128;
+}
+```
+
+It equivalent to the following code:
+
+```rust
+mod ext_calculator {
+    pub fn mult(a: U64, b: U64, receiver_id: &AccountId, deposit: Balance, gas: Gas) -> Promise {
+        Promise::new(receiver_id.clone())
+            .function_call(
+                b"mult",
+                json!({ "a": a, "b": b }).to_string().as_bytes(),
+                deposit,
+                gas,
+            )
+    }
+
+    pub fn sum(a: U128, b: U128, receiver_id: &AccountId, deposit: Balance, gas: Gas) -> Promise {
+        // ...
+    }
+}
+```
+
+Let's assume the calculator is deployed on `calc.near`, we can use the following:
+
+```rust
+const CALCULATOR_ACCOUNT_ID: &str = "calc.near";
+const NO_DEPOSIT: Balance = 0;
+const BASE_GAS: Gas = 5_000_000_000_000;
+
+#[near_bindgen]
+impl Contract {
+    pub fn sum_a_b(&mut self, a: U128, b: U128) -> Promise {
+        let calculator_account_id: AccountId = CALCULATOR_ACCOUNT_ID.to_string();
+        ext_calculator::sum(a, b, &calculator_account_id, NO_DEPOSIT, BASE_GAS)
+    }
+}
+```
+
+## Reuse crates from `near-sdk`
+
+`near-sdk` re-exports the following crates:
+
+- `borsh`
+- `base64`
+- `bs58`
+- `serde`
+- `serde_json`
+- `wee_alloc` (Though you will likely use the `setup_alloc` macro instead of importing it directly)
+
+Most common crates include `borsh` that is needed for internal STATE serialization and
+`serde` for the external JSON serialization.
+
+When marking structs with `serde::Serialize` you need to use `#[serde(crate = "near_sdk::serde")]`
+to point serde into the correct base crate.
+
+```rust
+/// Import `borsh` from `near_sdk` crate 
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+/// Import `serde` from `near_sdk` crate 
+use near_sdk::serde::{Serialize, Deserialize};
+
+/// Main contract structure serialized with Borsh
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize, PanicOnDefault)]
+pub struct Contract {
+    pub pair: Pair,
+}
+
+/// Implements both `serde` and `borsh` serialization.
+#[derive(Serialize, Deserialize, BorshDeserialize, BorshSerialize)]
+#[serde(crate = "near_sdk::serde")]
```

Comment:
Is it common to implement both? If this is not common, can we change the example to something that is? If it is common, can we explain why it's useful?

Reply:
added a comment about serde being useful

Reply:
Great info about serde, thanks.

I'm also curious about using both "Serialize/Deserialize" and "BorshSerialize/BorshDeserialize."

Reply:
Sure, feel free to make an issue. Here are some helpful links:
- https://borsh.io/
- https://docs.near.org/docs/concepts/storage-staking#use-a-binary-serialization-format-rather-than-json